iwyu_output.cc

```diff
@@ -605,6 +619,29 @@ void IwyuFileInfo::ReportForwardDeclareUse(SourceLocation use_loc,
   LogSymbolUse("Marked fwd-decl use of decl", symbol_uses_.back());
 }
 
+void IwyuFileInfo::ReportUsingDeclUse(SourceLocation use_loc,
+                                      const UsingDecl* using_decl,
+                                      bool in_cxx_method_body,
+                                      const char* comment) {  
+
+  // If accessing a symbol through a using decl in the same file that contains
+  // the using decl, we must mark the using decl as referenced. At the end of
+  // traversing the AST, we check to see if a using decl is unreferenced and a
+  // full use of one of its shadow decls so that the source file continues to
+  // compile.
+  typename map<const UsingDecl*, bool>::iterator using_decl_status = 
+    using_decl_referenced_.find(using_decl);
+
+  if(using_decl_status != using_decl_referenced_.end()) {
```

Comment:
In the last review, we hypothesized that this should be a hard error if the using decl is not found.  This actually isn't the case.  Here, we are looking to see if a using decl is referenced in the *same* file it is declared in.  Only then do we care to mark the using decl as referenced.  This is strictly to cover the case where a header file exports a symbol though a using decl but doesn't actually use the using decl at all.